cht-petals/setup-petals.sh

```diff
@@ -0,0 +1,104 @@
+#!/bin/bash
```

Comment:
I think this file belongs in `prem-app`, not here :)

Reply:
The installation part could be moved to prem-app so that we have a single file for both consumer and provider (swarm mode), but the last part of the file (run_petals_service) should be here. Prem App rust controller will fetch this file from raw.github.com and run it accordingly with bash.

Reply:
yea exactly.

This file here should:

- start from the clone repo step
- use env vars like `${PREM_APPDIR:-.}`, `${PREM_PYTHON:-python}` etc

Reply:
should we move it to the swarm-mode PR remove the sidecar and use Python env?

\cc @filopedraz @casperdcl

Reply:
Yep, seems the only solution. What do you think @casperdcl? @nsosio please sync with @biswaroop1547. Some code pairing could help

Reply:
sure, lemme break down the script into two parts:
- Env installation which will be part of prem-app
- repo setup (by using env vars set by above script mostly)
Will keep the second script only in this repo.

Reply:
Also I want us to agree on what env vars name to use (to be consistent), currently I care about two:
- `PREM_APPDIR`: everything related to python/miniconda env will be set inside this dir path.
- `PREM_ENV`: name of the conda env to use (looks important as there might be cases where we want to later manage multiple conda envs)



are we okay to go with above two names?
Also does anyone forsee any other var here?

Reply:
I think swarm-mode should support both pyinstaller and bash (even if one is not quite working yet). It's fine to implement in 2 separate PRs though.

Reply:
env vars: I think `PREM_PYTHON` (path to a python binary) rather than `PREM_ENV`. prem-services should not be at all aware of venv/virtualenv/conda/etc or any env management. It should only do `${PREM_PYTHON:-python} -m pip install -r requirements.txt`.

Reply:
thanks! that makes sense, so only these for now:
- `PREM_APPDIR`
- `PREM_PYTHON`

Reply:
made a lot of changes; please check if it works @biswaroop1547 :)